File: src/Sample/Plumix.Sample/Demos/General/BaselineDemoPage.cs
using Avalonia;
using Avalonia.Media;
using Plumix.Rendering;
using Plumix.UI;
using Plumix.Widgets;

namespace Plumix;

// Dart parity source: dart_sample/lib/demos/general/baseline_demo_page.dart (exact sample parity)

public sealed class BaselineDemoPage : StatelessWidget
{
    public override Widget Build(BuildContext context)
    {
        return new Column(
            crossAxisAlignment: CrossAxisAlignment.Stretch,
            spacing: 12,
            children:
            [
                new Text("Baseline + IgnoreBaseline", fontSize: 20, color: Colors.Black),
                new Text(
                    "The guide is 48 px from the top. Text uses its real alphabetic baseline; the box falls back " +
                    "to its bottom edge.",
                    fontSize: 14,
                    color: Colors.DimGray),
                BuildBaselinePreview(),
                new Text(
                    "IgnoreBaseline keeps the tall middle child out of Row baseline calculations.",
                    fontSize: 14,
                    color: Colors.DimGray),
                new Container(
                    color: Color.Parse("#FFF1F5F9"),
                    padding: new Thickness(12),
                    child: new Row(
                        crossAxisAlignment: CrossAxisAlignment.Baseline,
                        textBaseline: TextBaseline.Alphabetic,
                        spacing: 12,
                        children:
                        [
                            new Text("Aa", fontSize: 34, color: Color.Parse("#FF1D3557")),
                            new IgnoreBaseline(
                                child: new Container(
                                    width: 32,
                                    height: 52,
                                    color: Color.Parse("#FFE9C46A"))),
                            new Text("baseline", fontSize: 16, color: Color.Parse("#FF2A9D8F")),
                        ])),
            ]);
    }

    private static Widget BuildBaselinePreview()
    {
        return new Container(
            height: 118,
            color: Color.Parse("#FFE7EDF6"),
            padding: new Thickness(12),
            child: new Stack(
                clipBehavior: Clip.None,
                children:
                [
                    new Positioned(
                        left: 0,
                        right: 0,
                        top: 48,
                        height: 1,
                        child: new Container(color: Color.Parse("#FFE63946"))),
                    new Baseline(
                        baseline: 48,
                        baselineType: TextBaseline.Alphabetic,
                        child: new Text("Plumix", fontSize: 36, color: Color.Parse("#FF1D3557"))),
                    new Positioned(
                        left: 150,
                        child: new Baseline(
                            baseline: 48,
                            baselineType: TextBaseline.Alphabetic,
                            child: new Container(
                                width: 54,
                                height: 28,
                                color: Color.Parse("#FF2A9D8F")))),
                ]));
    }
}

File: dart_sample/lib/demos/general/baseline_demo_page.dart
import 'package:material_ui/material_ui.dart';

class BaselineDemoPage extends StatelessWidget {
  const BaselineDemoPage({super.key});

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 12,
      children: <Widget>[
        const Text(
          'Baseline + IgnoreBaseline',
          style: TextStyle(fontSize: 20, color: Colors.black),
        ),
        const Text(
          'The guide is 48 px from the top. Text uses its real alphabetic baseline; '
          'the box falls back to its bottom edge.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        _buildBaselinePreview(),
        const Text(
          'IgnoreBaseline keeps the tall middle child out of Row baseline calculations.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        Container(
          color: const Color(0xFFF1F5F9),
          padding: const EdgeInsets.all(12),
          child: Row(
            crossAxisAlignment: CrossAxisAlignment.baseline,
            textBaseline: TextBaseline.alphabetic,
            spacing: 12,
            children: <Widget>[
              const Text(
                'Aa',
                style: TextStyle(fontSize: 34, color: Color(0xFF1D3557)),
              ),
              IgnoreBaseline(
                child: Container(
                  width: 32,
                  height: 52,
                  color: const Color(0xFFE9C46A),
                ),
              ),
              const Text(
                'baseline',
                style: TextStyle(fontSize: 16, color: Color(0xFF2A9D8F)),
              ),
            ],
          ),
        ),
      ],
    );
  }

  static Widget _buildBaselinePreview() {
    return Container(
      height: 118,
      color: const Color(0xFFE7EDF6),
      padding: const EdgeInsets.all(12),
      child: Stack(
        clipBehavior: Clip.none,
        children: <Widget>[
          Positioned(
            left: 0,
            right: 0,
            top: 48,
            height: 1,
            child: Container(color: const Color(0xFFE63946)),
          ),
          const Baseline(
            baseline: 48,
            baselineType: TextBaseline.alphabetic,
            child: Text(
              'Plumix',
              style: TextStyle(fontSize: 36, color: Color(0xFF1D3557)),
            ),
          ),
          Positioned(
            left: 150,
            child: Baseline(
              baseline: 48,
              baselineType: TextBaseline.alphabetic,
              child: Container(
                width: 54,
                height: 28,
                color: const Color(0xFF2A9D8F),
              ),
            ),
          ),
        ],
      ),
    );
  }
}
